pinc/Round.inc

```diff
@@ -1,20 +1,55 @@
 <?php
+include_once($relPath.'Stage.inc');
 
-// $Id$
+/*
+ * A singleton container class for accessing Round objects.
+ * See the Activities class definition for more information.
+ */
+class Rounds extends Stages
```

Comment:
This will be more complicated than Activity but could use the trait with an override like
```
class Rounds
{
    use StatObj {set as obj_set;}

    private static $round_for_number = [];
    private static $round_for_page_state = [];
    private static $page_states = [];

    public static function set($object)
    {
        self::obj_set($object);
        self::$round_for_number[$object->round_number] = $object;
    }

    public static function set_page_state($page_state, $object)
    {
        self::$page_states[] = $page_state;
        self::$round_for_page_state[$page_state] = $object;
    }

    public static function get_page_states()
    {
        return self::$page_states;
    }

    public static function get_for_page_state($page_state)
    {
        return self::$round_for_page_state[$page_state] ?? null;
    }

    public static function get_for_number($number)
    {
        return self::$round_for_number[$number] ?? null;
    }
}
```

Reply:
I'm uncertain that accesses by number, page_state, etc are oft-used enough that we care about using an assoc array for them. We'll probably spend more time initializing them for every page load than we do using them after they are initialized.

I'll add some logging to some of the current access functions and see how often they are used on average, but my intuition is that they aren't used enough to be worth the trouble of creating an assoc array.